pkgs/build-support/trivial-builders/default.nix

```
@@ -322,22 +333,18 @@ rec {
     };
 
   /*
-    Similar to writeShellScriptBin and writeScriptBin.
-    Writes an executable Shell script to /nix/store/<store path>/bin/<name> and
-    checks its syntax with shellcheck and the shell's -n option.
-    Individual checks can be foregone by putting them in the excludeShellChecks
-    list, e.g. [ "SC2016" ].
-    Automatically includes sane set of shellopts (errexit, nounset, pipefail)
-    and handles creation of PATH based on runtimeInputs
-
-    Note that the checkPhase uses stdenv.shell for the test run of the script,
-    while the generated shebang uses runtimeShell. If, for whatever reason,
-    those were to mismatch you might lose fidelity in the default checks.
+    Similar to `writeShellScriptBin` and `writeScriptBin`.
```

Maybe better:

```suggestion
    Similar to [](#trivial-builder-writeShellScriptBin) and [](#trivial-builder-writeScriptBin).
```

This will render sensibly in HTML.

---

This kind of directly contradicts what I said in https://github.com/NixOS/nixpkgs/pull/280592#discussion_r1452760334 :sweat_smile: 

I guess you meant that if this content is moved to `trivial-build-helpers.chapter.md`, it would get rendered correctly.

---

OK, updated the manual. I'll leave the argument documentation where it is, so it'll get rendered when nixdoc gets wired up.

---

I guess it's good to have docs at all, so I don't want to block on this, but really it would be best if the docs were actually visible from the manual instead of people having to look through the source all the time.

Also I want to highlight that @DanielSidhion is recently writing some really nice rendered documentation for function arguments: https://github.com/NixOS/nixpkgs/blob/7d021ca26b35606f3b0382b8811c51c0b65aef30/doc/build-helpers/images/dockertools.section.md#L28

Rendered: https://nixos.org/manual/nixpkgs/unstable/#ssec-pkgs-dockerTools-buildImage-inputs